packages/next/lib/resolve-request.ts

```diff
@@ -8,6 +8,17 @@ export function resolveRequest(req: string, issuer: string): string {
   if (process.versions.pnp) {
     const { resolveRequest: pnpResolveRequest } = require(`pnpapi`)
     return pnpResolveRequest(req, issuer, { considerBuiltins: false })
+  } else if (process?.env?.npm_config_user_agent?.startsWith('pnpm/')) {
+    // fixes webpack builds when using pnpm by switching to the default node resolution algorithm
+    // see: preserveSymlinks
+    // from : https://www.npmjs.com/package/resolve
+    //  - Note: this property is currently true by default but it will be changed to false in the next major
+    //           version because Node's resolution algorithm does not preserve symlinks by default.
+    const basedir =
+      issuer.endsWith(path.posix.sep) || issuer.endsWith(path.win32.sep)
+        ? issuer
+        : path.dirname(issuer)
+    return resolve.sync(req, { basedir, preserveSymlinks: false })
```

Comment:
We can also consider moving  preserveSymlinks: false  as default for all package managers and not have this conditional just for pnpm

Reply:
FWIW, this is the approach I took in https://github.com/vercel/next.js/pull/16369